File: Leetcode/Python/_1351.py
class Solution:    
    def countNegatives(self, grid: List[List[int]]) -> int:
        def countNegativeInRow(row: List[int]) -> int:
            left = 0
            right = len(row) - 1

            while left <= right:
                mid = left + (right - left) // 2
                if row[mid] >= 0:
                    left = mid + 1
                else:
                    right = mid - 1
            return len(row) - left
        
        count = 0
        for idx in range(len(grid)):
            count += countNegativeInRow(grid[idx])
        return count

File: Leetcode/cpp/1351.cpp
class Solution {
public:
  int binsearch(vector<int>& array) {
    int left = -1, right = array.size(), mid, len = array.size();
    while (left < right - 1) {
      int m = left + (right - left) / 2;
      if (array[m] >= 0) {
        left = m;
      } else {
        right = m;
      }
    }
    return len - left - 1;
  }
  int countNegatives(vector<vector<int>>& grid) {
    int count = 0;
    for (int i = 0; i < grid.size(); i++) {
      count += binsearch(grid[i]);
    }
    return count;
  }
};
